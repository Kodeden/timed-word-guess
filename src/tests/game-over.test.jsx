import { render, screen } from "@testing-library/react";
import GameOver from "../components/game-over";

it("renders 'You won!' when the status is 'won'", () => {
  render(<GameOver status="won" />);

  expect(screen.getByText(/won/i, { selector: "h2" })).toBeInTheDocument();
});

it("renders 'You lost!' when the status is 'lost'", () => {
  render(<GameOver status="lost" />);

  expect(screen.getByText(/lost/i, { selector: "h2" })).toBeInTheDocument();
});
