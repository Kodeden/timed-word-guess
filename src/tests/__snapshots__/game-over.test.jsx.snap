// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renders without crashing 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="container mx-auto flex flex-col justify-center"
      >
        <h2
          class="text-green-400 my-2 text-center text-2xl font-black capitalize "
        >
          You 
          won
          !
        </h2>
        <img
          alt="won"
        />
        <button
          class="button"
        >
          Shall we play again?
        </button>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="container mx-auto flex flex-col justify-center"
    >
      <h2
        class="text-green-400 my-2 text-center text-2xl font-black capitalize "
      >
        You 
        won
        !
      </h2>
      <img
        alt="won"
      />
      <button
        class="button"
      >
        Shall we play again?
      </button>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
